src/main/java/gillycarpetaddons/mixins/FlowerBlock_replaceFlowersMixin.java

```diff
@@ -0,0 +1,38 @@
+package gillycarpetaddons.mixins;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.FlowerBlock;
+import net.minecraft.block.Material;
+import net.minecraft.item.ItemPlacementContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+
+import static gillycarpetaddons.GillyCarpetAddonsSettings.replaceableFlowers;
+
+@Mixin(FlowerBlock.class)
+public abstract class FlowerBlock_replaceFlowersMixin extends Block {
+
+  //makes ide happy
+  FlowerBlock_replaceFlowersMixin() {
+    super(Settings.of(Material.PLANT));
+  }
+
+  //overrides canReplace
+  public boolean canReplace(BlockState state, ItemPlacementContext context) {
+    if (!replaceableFlowers) {
+      return super.canReplace(state, context);
+    }
+    return context.getStack().isEmpty() || !context.getStack().isOf(this.asItem());
```

Comment:
I'm not entirely sure how this determines whether we can replace the flower?

I get the last bit, determining that the item we're replacing it with is not a flower, but not the empty stack context?

What if the item is not a block?

Reply:
that is a very much a concern i over looked

Reply:
This is a ckeck from AbstrackBlock.canReplace() with Material.isReplacable removed for obvius reasons.